Fix several typos

jdufresne opened the pull request.

deivid-rodriguez reviewed (commented):
Thanks!

Can you contribute fixes to `bundler/lib/bundler/vendor/thor/` files to the [upstream repository](https://github.com/rails/thor)?

Also, how did you find all these? We used to have a spell checker in CI, but it never found any issues and I removed it.

jdufresne commented:
> Can you contribute fixes to bundler/lib/bundler/vendor/thor/ files to the [upstream repository](https://github.com/rails/thor)?

Yup! I'll work on moving those upstream.

I have removed them from the PR here so this only includes non-vendored files.

> Also, how did you find all these? We used to have a spell checker in CI, but it never found any issues and I removed it.

I used [codespell](https://github.com/codespell-project/codespell). I believe Rails also uses this tool.

jdufresne commented:
Thor PR opened at:
https://github.com/rails/thor/pull/806

simi commented:
Should we onboard this tool as well?

https://github.com/rails/rails/blob/c49b8270e18375d64a045da2662f4d31e23726cf/.github/workflows/lint.yml#L25-L42

deivid-rodriguez commented:
Definitely!

For what it's worth, we did have [spell checking in CI](https://github.com/rubygems/rubygems/pull/4202) for about a year, but [I removed it](https://github.com/rubygems/rubygems/commit/f1bd461ac7d77901099c0a3b7e2113e82a90d362) because the one time the CI check showed as failed, it was because the action failed to install, not because of detecting any typos.

This one seems very easy to setup: https://github.com/codespell-project/actions-codespell. We could try if it detects a fair amount of the typos being fixed here.

jdufresne commented:
I went ahead and added a GitHub actions for codespell. This should catch future misspellings.

simi commented on .codespellignore:
```diff
@@ -0,0 +1,14 @@
+filetest
+fo
+hax
+msdos
```
:older_adult: :)

simi commented on .github/workflows/spell.yml:
```diff
@@ -0,0 +1,19 @@
+    name: Check spelling
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Can you lock both actions to given SHA with tag comment like it is done at

https://github.com/rubygems/rubygems/blob/75ec11612de0818c1efb83db017484c65b2c8ea4/.github/workflows/daily-bundler.yml#L23

:question:

jdufresne replied:
Sure thing. Done in the latest push.

deivid-rodriguez reviewed (approved):
Thanks so much, this is great!
